### PR TITLE
Bluespace Shakers

### PR DESCRIPTION
**Pull request by Cisrox**
## What this does
Adds the bluespace shaker, which is a ~~200u~~ 300u variant of the shaker. ~~The tech requirements for this are identical to the bluespace beaker.~~ The tech requirements for this are equal to the large stasis beaker, the material costs are higher. 

proof i tested:

test left
test right
test shake2
test shake
test flip
science test
test destruct

## Why it's good
Allows bartenders to make bigger jungle juice concoctions.

## Changelog
:cl:

 * rscadd: Adds bluespace shakers, which can be created in RnD.
 * imageadd: Adds new bluespace shaker sprites. (Falcon2346)

### CODE REVIEW

**Comment by Dacendeth** [comment]
It should have the same price/tech levels as a large stasis beaker otherwise there is literally no point to craft either stasis beaker as they're just worse and/or more expensive.

**Comment by jwhitak** [comment]
I'll thumbs up once Dacendeth's concerns are answered: these shakers should be compared to stasis beakers, not bluespace beakers. The no react flag is powerful.

**Comment by Eneocho** [comment]
Large Bluespace Shaker (300u) when???

**Comment by SonixApache** [comment]
prepare for unforeseen consequences

File: code/modules/reagents/reagent_containers/food/drinks.dm [hunk 1]
```diff
@@ -1359,6 +1359,63 @@
...
+
+/obj/item/weapon/reagent_containers/food/drinks/bluespaceshaker
+	name = "\improper bluespace shaker"
+	desc = "A bluespace shaker to mix drinks in."
+	icon_state = "bluespaceshaker"
+	origin_tech = Tc_BLUESPACE + "=4;" + Tc_MATERIALS + "=6"
+	starting_materials = list(MAT_GLASS = 2000)
+	w_type = RECYK_GLASS
+	amount_per_transfer_from_this = 10
+	volume = 300
```
Comment (SonixApache) [inline code/modules/reagents/reagent_containers/food/drinks.dm:1451]: ?

Comment (Cisrox) [reply code/modules/reagents/reagent_containers/food/drinks.dm:1451]: starting_materials: the amount is identical to bluespace beakers (what i originally based the material cost on). this variable is the amount of materials you get for directly inserting the item into a protolathe.
w_type: the type of material that it gets recycled into in the waste area of the station

image
educate

Comment (Cisrox) [reply code/modules/reagents/reagent_containers/food/drinks.dm:1451]: will update this to more accurately reflect the cost now that i changed it, i had forgotten about this

**Comment by Cisrox** [comment]
if you have any genuine concerns other than "new thing bad" please tell me, I had a conversation with someone about the power of stasis beakers, and there is really no reason to not give bartenders 300u to play with.

**"300u reaction on command is OP!"**: you can literally do this but multiple times over with grenades

**"300u will let people spam reactions too much!"**: if you have the diamonds to make stasis beakers/these, you can literally just make 2 large stasis beakers which would have more space to fuck around with.

**"300u splash reaction OP!"**: you can't splash with shakers, they're a subset of drinking glasses. if you aim someone with harm intent you will still attempt to make them drink it.

**"300u of reactions for cryo gimmicks"**: you can't put shakers in cryo

bartenders will have another reason to go to science (other than getting weird mats like uranium for certain drinks, which is kinda rare), they can make big batches of whatever complex ass drinks they are trying to make. bartenders will be around science a little bit more so they can ask for other shit when they're there/get to know the social outcasts who don't leave RnD

File: code/modules/reagents/reagent_containers/food/drinks.dm (hunk 1 above)
Comment (SonixApache) [reply code/modules/reagents/reagent_containers/food/drinks.dm:1451]: your OP stuff says 200u, the description text in the other file says 200u, but your code says 300u
just lower this back to 200u

**Comment by SonixApache** [comment]
> if you have any genuine concerns other than "new thing bad" please tell me, I had a conversation with someone about the power of stasis beakers, and there is really no reason to not give bartenders 300u to play with.
> 
> **"300u reaction on command is OP!"**: you can literally do this but multiple times over with grenades
> 
> **"300u will let people spam reactions too much!"**: if you have the diamonds to make stasis beakers/these, you can literally just make 2 large stasis beakers which would have more space to fuck around with.
> 
> **"300u splash reaction OP!"**: you can't splash with shakers, they're a subset of drinking glasses. if you aim someone with harm intent you will still attempt to make them drink it.
> 
> **"300u of reactions for cryo gimmicks"**: you can't put shakers in cryo
> 
> bartenders will have another reason to go to science (other than getting weird mats like uranium for certain drinks, which is kinda rare), they can make big batches of whatever complex ass drinks they are trying to make. bartenders will be around science a little bit more so they can ask for other shit when they're there/get to know the social outcasts who don't leave RnD

large stasis beakers hold 200u, this holds 300u, that's weird powercreep and that's all really
stating "oh but you could just get multiple to do the same thing" is a bad comparison; you can hold a larger amount of reagents in this particular one-item portable hidable noreact vessel and that's odd

**Comment by Cisrox** [comment]
>> large stasis beakers hold 200u, this holds 300u, that's weird powercreep and that's all really stating "oh but you could just get multiple to do the same thing" is a bad comparison; you can hold a larger amount of reagents in this particular one-item portable hidable noreact vessel and that's odd

Bro I literally explained how it isn’t powercreep.

I made it 200u initially because I had intended it to be greater than the large stasis beaker, but after reviewing the material costs I realized large stasis beakers are already 200u. So I upped it to 300u. If you can give me an actual, concrete, solid, explained reason as to HOW this is powercreep other than saying “UNFORESEEN CONSEQUENCES!!” I would be open to lowering it. 


Just because it is more units doesn’t necessarily mean it is more powerful. You cannot do anything powerful with the extra units that you can’t already do with a large stasis beaker. You can’t splash with them, so you can’t dump 300u reaction on someone from your pocket. That is the most powerful usage stasis beakers have regarding volume. If you want to spam reactions by sipping it or something, you can literally already do that with stasis beakers. 

If you, for some reason, are trying to sip something 10000 times, there is literally no difference as to whether you use this or a large stasis beaker. You can fit both in the same places, you could fit like 20 stasis beakers in your backpack in a box. If you are using this, you have access to science with full mats, which you can literally duplicate materials and don’t have to worry about cost. You could make a bag of holding and fit hundreds in there.

The regular one doesn’t even require diamonds. Only the large stasis beaker does.

Giving bartenders space to play around and make their job more fun isn’t a crime man

**Comment by SonixApache** [comment]
\>this thing does the same thing as other things of its type, except its stronger
\>no, that's not powercreep
brother please

bartenders will be happy with their 200u, they don't need 300u

File: code/modules/reagents/reagent_containers/food/drinks.dm (hunk 1 above)
Comment (wotisjanitor) [reply code/modules/reagents/reagent_containers/food/drinks.dm:1451]: ?

**Comment by Eneocho** [comment]
Can't wait for 25u noreact glasses so you can serve people glasses of smoke.

**Review by Kurfursten — CHANGES_REQUESTED** [review]
You need to implement this using inheritance instead of copy/pasting shaker code. Your implementation of the shaker should be only like five lines long, just changing the name, description, icon, and volume.

**Comment by Cisrox** [comment]
yeah i did this very lazily, will fix it

**Comment by Kurfursten** [comment]
But will you?

**Comment by ShiftyRail** [comment]
He didn't fix it; also conflicts.

**Comment by Cisrox** [comment]
I fixed the inheritance but haven't committed that yet. I did find a bug where the shaker wouldn't mix it's contents properly after I was testing my changes which I still need to fix. Shaker bros its not over yet

**Comment by Kurfursten** [comment]
(It was actually over)

**Comment by Kurfursten** [comment]
You have until the end of the year to finish this or it will be considered derelict!

**Comment by Eneocho** [comment]
"You have until the end of the year to finish this or it will be considered derelict!"
image
LITERALLY 1984

**Comment by Kurfursten** [comment]
Actually it will be 2024

**Comment by Eneocho** [comment]
@Cisrox quick make an update before you get collab'd

**Review by Kurfursten — CHANGES_REQUESTED** [review]
It would appear that you forgot to actually include the code for the new shaker. There's only code for the design datum.

**Comment by Cisrox** [comment]
> It would appear that you forgot to actually include the code for the new shaker. There's only code for the design datum.

I restarted to fix the conflicts and fell asleep after copying the other stuff over

**Comment by Cisrox** [comment]
This is functionally complete and tested now, the only thing that needs to be fixed before this gets merged is finding a less hacky solution for a bug where only 100u gets mixed at a time due to the way that shakers react their contents

**Comment by Kurfursten** [comment]
Just increase the size of the reaction container to be 300u
This should be fine because the normal shaker can't dump more than 100 into it anyway

**Comment by Cisrox** [comment]
bet, i merged that before i left for new years partying last night, codewise everything should be good. in a couple hours when i get home i’m gonna run through all the features just to make sure nothing accidentally broke, after which it should be good to merge

**Comment by Cisrox** [comment]
it works
IMG_1805

**Review by Kurfursten — APPROVED** [review]

**Pull request merged**